scripts/install_backend_python_libs.py

```diff
@@ -0,0 +1,276 @@
+    current 'requirements.txt' file requirements. It takes a list of mismatches
+    and corrects those mismatches by installing or uninstalling packages.
+
+    Args:
+        mismatches: dict(string, tuple(string|None, string|None)). Dictionary
+            with the library names as keys and a tuple as values. The 1st
+            element of the tuple is the version string of the library required
+            by the requirements.txt file while the 2nd element is the version
+            string of the library currently installed in the
+            'third_party/python_libs' directory. If the library doesn't exist,
+            the tuple element will be None. For example, this dictionary
+            signifies that 'requirements.txt' requires flask with version 1.0.1
+            while the 'third_party/python_libs' directory contains flask 1.1.1:
+                {
+                  flask: ('1.0.1', '1.1.1')
+                }
+    """
+    # Handling 5 or more mismatches requires 5 or more individual `pip install`
+    # commands which is slower than just reinstalling all of the libraries using
+    # `pip install -r requirements.txt`.
+    if len(mismatches) >= 5:
+        if os.path.isdir(common.THIRD_PARTY_PYTHON_LIBS_DIR):
+            shutil.rmtree(common.THIRD_PARTY_PYTHON_LIBS_DIR)
+        subprocess.check_call([
+            'pip', 'install', '--target',
+            common.THIRD_PARTY_PYTHON_LIBS_DIR,
+            '--no-dependencies', '-r',
+            common.REQUIREMENTS_FILE_PATH
+        ])
+        return
+
+    for library, versions in mismatches.items():
+        requirements_version = (
+            pkg_resources.parse_version(versions[0]) if versions[0] else None)
+        directory_version = (
+            pkg_resources.parse_version(versions[1]) if versions[1] else None)
+
+        # The library is installed in the directory but is not listed in
+        # requirements.
+        if not requirements_version:
+            shutil.rmtree(common.THIRD_PARTY_PYTHON_LIBS_DIR)
+            subprocess.check_call([
+                'pip', 'install', '--target',
+                common.THIRD_PARTY_PYTHON_LIBS_DIR,
+                '--no-dependencies', '-r',
+                common.REQUIREMENTS_FILE_PATH
+            ])
+
+        # Either the library listed in 'requirements.txt' is not in the
+        # third party directory or the currently installed library version is
+        # not up-to-date with the required 'requirements.txt' version.
+        elif (not directory_version or
+              requirements_version > directory_version):
+            subprocess.check_call([
+                'pip', 'install', '--target',
+                common.THIRD_PARTY_PYTHON_LIBS_DIR,
+                '--no-dependencies',
+                '%s==%s' % (
+                    library,
+                    python_utils.convert_to_bytes(requirements_version)),
+                '--upgrade'
```

Comment:
Just to check -- this deletes/replaces old metadata files for the lib in question, right?